Spec/Auth.swift

```diff
@@ -2091,61 +2091,283 @@ class Auth : QuickSpec {
+                            expect(tokenDetails).to(beNil())
+                            expect(authCallbackHasBeenInvoked).to(beFalse())
+                            expect(rest.auth.options.authCallback).to(beNil())
+                            done()
+                        }
+                    }
+                }
+
+                fit("should supersede configured params and options even if arguments are empty") {
+                    let options = AblyTests.clientOptions()
+                    options.key = "xxxx:xxxx"
+                    options.clientId = "client_string"
+                    let rest = ARTRest(options: options)
+
+                    let tokenParams = ARTTokenParams(clientId: options.clientId)
+                    let defaultTtl = tokenParams.ttl
+                    let defaultCapability = tokenParams.capability
+
+                    // Defaults
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(nil, options: nil) { tokenDetails, error in
+                            guard let error = error else {
+                                fail("Error is nil"); done(); return
+                            }
+                            expect(error.code).to(equal(40400))
+                            expect(tokenDetails).to(beNil())
+                            done()
+                        }
+                    }
+
+                    // Custom
+                    tokenParams.ttl = ExpectedTokenParams.ttl
+                    tokenParams.capability = ExpectedTokenParams.capability
+                    tokenParams.clientId = nil
+
+                    let authOptions = ARTAuthOptions()
+                    authOptions.key = AblyTests.commonAppSetup().key
+                    authOptions.queryTime = true
+
+                    var serverDate = NSDate()
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.time { date, error in
+                            expect(error).to(beNil())
+                            guard let date = date else {
+                                XCTFail("No server time"); done(); return
+                            }
+                            serverDate = date
+                            done()
+                        }
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(tokenParams, options: authOptions) { tokenDetails, error in
+                            expect(error).to(beNil())
+                            guard let tokenDetails = tokenDetails else {
+                                XCTFail("TokenDetails is nil"); done(); return
+                            }
+                            expect(tokenDetails.clientId).to(beNil())
+                            expect(tokenDetails.issued).to(beCloseTo(serverDate, within: 1.0)) //1 Second
```

Comment:
Plus, anything could take longer that 1 second in the server for any reason and then `serverDate` would be outdated, especially given you're using it twice. I'd rather not introduce even more reasons for tests to randomly fail because we expect unguaranteed things from the server.